### DISCUSSION TITLE
Errors from versions 1.24.1 -> 1.25.1

### DISCUSSION THREAD

**Discussion by tsmeets2** [AzureAD/microsoft-identity-web#1797] [Q&A]
Hi I am struggling with a log problem and I hope you guys can help me.  

I've updated from version 1.24.1 to 1.25.1
Everything works but I get the errors below. Is this some sort of false positive? What is happening here?
My steps:
1. I request a token. 
2. the token is send to a 3rd party
3. from there the token is used to make a call.

The 3 steps work. They just work. If I use a invalid token it does not work. This is also perfect. Has someone an Idea to why these errors trigger from 1.25.1?

Errors:
IDX10503: Signature validation failed. Token does not have a kid. Keys tried........

IDX10634: Unable to create the SignatureProvider.
Algorithm: 'HS512', SecurityKey: 'Microsoft.IdentityModel.Tokens.X509SecurityKey, KeyId: 'OzZ5Dbmcso9Qzt2ModGmihg30Bo', InternalId: 'OzZ5Dbmcso9Qzt2ModGmihg30Bo'.'
 is not supported. The list of supported algorithms is available here: https://aka.ms/IdentityModel/supported-algorithms

IDX10634: Unable to create the SignatureProvider.
Algorithm: 'HS512', SecurityKey: 'Microsoft.IdentityModel.Tokens.RsaSecurityKey, KeyId: 'DqUu8gf-nAgcyjP3-SuplNAXAnc', InternalId: '8cTdfkGpvGb-S7Jm4j2Qx_vFGaTWhyHvzyUNxDldt00'.'
 is not supported. The list of supported algorithms is available here: https://aka.ms/IdentityModel/supported-algorithms

my configuration:

            // Adding Authentication  
            services.AddAuthentication( options =>
            {
                options.DefaultAuthenticateScheme = JwtBearerDefaults.AuthenticationScheme;
                options.DefaultChallengeScheme = JwtBearerDefaults.AuthenticationScheme;
                options.DefaultScheme = JwtBearerDefaults.AuthenticationScheme;
            } )
            // Adding Jwt Bearer  
            .AddJwtBearer( options =>
            {
                options.SaveToken = true;
                options.RequireHttpsMetadata = false;
                options.TokenValidationParameters = new TokenValidationParameters()
                {
                    ValidateIssuer = true,
                    ValidateAudience = true,
                    ValidAudience = Configuration[ "JWT:ValidAudience" ],
                    ValidIssuer = Configuration[ "JWT:ValidIssuer" ],
                    IssuerSigningKey = new SymmetricSecurityKey( Encoding.UTF8.GetBytes( Configuration[ "JWT:Secret" ] ) )
                };
            } )
            .AddMicrosoftIdentityWebApi( Configuration.GetSection( "AzureAd" ), "AzureAD" );

            services.AddAuthorization( options =>
            {
                options.DefaultPolicy = new AuthorizationPolicyBuilder( JwtBearerDefaults.AuthenticationScheme, "AzureAD" )
                    .RequireAuthenticatedUser()
                    .Build();

                options.AddPolicy( "APIAccess", policy => policy.Requirements.Add( new ScopesRequirement( "Access.API" ) ) );
            } );

**Comment by jmprieur** [comment]
@tsmeets2 : 

1. why do you need the 
   ```CSharp
   .AddJwtBearer( options =>
   ```
   
   block?

   `AddMicrosoftIdentityWebApi` does the same?

2. Why do you provide the StS signing keys?

**Comment by tsmeets2** [comment]
Because I need it for another authentication.

We have 2 ways of authentication
1. Azure AD
2. normal tokens through username password (out of Active Directory)

Keep in mind, one is wih Bearer and the other is with AzureAd registered

To answer your second question, Why do you provide the StS signing keys?
IDX10503 and IDX10634 pops up in logging as errors. However everything works as expected. No problems at all.
